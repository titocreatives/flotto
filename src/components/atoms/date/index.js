import Date from "./date.vue";
export default Date;
