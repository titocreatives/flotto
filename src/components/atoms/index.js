export { default as Logo } from "./logo";
export { default as Date } from "./date";
