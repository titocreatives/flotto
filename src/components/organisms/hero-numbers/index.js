import HeroNumbers from "./hero-numbers.vue";
export default HeroNumbers;
