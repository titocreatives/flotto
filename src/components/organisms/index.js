export { default as HeroNumbers } from "./hero-numbers";
export { default as Layout } from "./layout";
