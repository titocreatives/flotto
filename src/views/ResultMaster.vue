<template>
  <div :class="`page result-${drawType}`">
    <section class="result-stage">
      <template v-if="!!drawData.draws">
        <div class="result-nav">
          <button
            @click="hazPast && drawsIndex--"
            :disabled="!hazPast"
            class="result-nav-prev"
          >
            ⟵
          </button>
          <Date :value="currentDraw.date" class="result-nav-date" />
          <button
            @click="hazMoah && drawsIndex++"
            :disabled="!hazMoah"
            class="result-nav-next"
          >
            ⟶
          </button>
        </div>
        <HeroNumbers v-bind="currentDraw" />
      </template>
      <template v-else>{{ "Whoopsy" + error }}</template>
    </section>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Date } from "@/components/atoms/index";
import { HeroNumbers } from "@/components/organisms/index";
import { drawData, drawQueryResult } from "@/types";
import { ApolloQueryResult } from "apollo-client";

export default Vue.extend({
  name: "ResultMaster",
  components: {
    Date,
    HeroNumbers
  },
  inject: ["$apollo"],
  data() {
    return {
      apollo: this.$apollo,
      drawData: {} as drawData,
      drawsIndex: 0,
      error: null
    };
  },
  props: {
    drawLimit: {
      type: Number,
      required: true
    },
    drawType: {
      type: String,
      required: true
    },
    mockData: [Boolean, Object]
  },
  methods: {
    getResults(): void {
      this.apollo
        .query({
          query: require("@/graphql/draw.gql"),
          variables: {
            limit: this.drawLimit,
            type: this.drawType
          }
        })
        .then(({ data: { draw } }: ApolloQueryResult<drawQueryResult>) => {
          this.drawData = draw;
        })
        .catch(e => {
          this.error = e.message;
          console.warn(e.message);
        })
        .finally(() => {
          this.$emit("slider-change", true);
        });
    }
  },
  computed: {
    hazMoah(): Boolean {
      return this.drawsIndex < this.drawData.draws.length - 1;
    },
    hazPast(): Boolean {
      return this.drawsIndex > 0;
    },
    currentDraw(): Object {
      return this.drawData.draws[this.drawsIndex];
    }
  },
  created(): void {
    if (this.mockData) {
      this.drawData = this.mockData;
    } else {
      this.getResults();
    }
  },
  beforeRouteLeave(to, from, next) {
    if (to.name === "home") {
      this.$emit("slider-change", false);
      setTimeout(() => next(), 450);
    }
  }
});
</script>

<style lang="scss">
@import "@/scss/index-for-components";

$gutter: 2%;

.result {
  &-nav {
    $width: 250px;

    @include position($space-micro, null, null, $gutter);

    width: $width;
    display: flex;
    justify-content: space-between;
    align-items: center;

    @include breakpoint(small only) {
      left: 50%;
      transform: translateX(-50%);
    }

    &-prev,
    &-next {
      transform: scale(1);

      &:hover:not([disabled]) {
        transform: scale(1.2);
      }
    }
  }

  &-stage {
    padding: $space-medium $gutter;
    background: $stage-gradient;

    @include breakpoint(medium up) {
      border-radius: $space-small;
    }

    @include breakpoint(small only){
      padding-bottom: 0;
    }
  }
}
</style>
