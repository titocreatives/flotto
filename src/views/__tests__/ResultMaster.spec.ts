import { mount } from "@vue/test-utils";
import ResultMaster from "./../ResultMaster.vue";
import { Date } from "@/components/atoms";
import { HeroNumbers } from "@/components/organisms";

const MOCK_DATA = {
  success: true,
  backendError: null,
  draws: [
    {
      date: "Dec 13 2019",
      numbers: ["25", "31", "38", "49", "50"],
      additionalNumbers: ["5", "10"]
    },
    {
      date: "Dec 06 2019",
      numbers: ["8", "14", "23", "30", "45"],
      additionalNumbers: ["1", "9"]
    },
    {
      date: "Nov 29 2019",
      numbers: ["2", "3", "30", "31", "45"],
      additionalNumbers: ["6", "8"]
    }
  ]
};

describe("Page ResultMaster", () => {
  let cmp: any;

  beforeAll(() => {
    cmp = mount(ResultMaster, {
      propsData: {
        drawLimit: 20,
        drawType: "eurojackpot",
        mockData: MOCK_DATA
      },
      provide: {
        $apollo: {}
      }
    });
  });

  it("matches snapshot", () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it("passes correct currentDraw data to atom Date and organism HeroNumbers", () => {
    const date = cmp.find(Date);
    const heroNumbers = cmp.find(HeroNumbers);

    expect(date.vm.value).toMatch(
      cmp.vm.currentDraw.date && MOCK_DATA.draws[0].date
    );
    expect(heroNumbers.vm.numbers).toEqual(
      cmp.vm.currentDraw.numbers && MOCK_DATA.draws[0].numbers
    );
    expect(heroNumbers.vm.additionalNumbers).toEqual(
      cmp.vm.currentDraw.additionalNumbers &&
        MOCK_DATA.draws[0].additionalNumbers
    );
  });

  it("can navigate results - next", () => {
    const next = cmp.find(".result-nav-next");
    next.trigger("click");
    expect(cmp.vm.drawsIndex).toBe(1);
  });

  it("disables next button", () => {
    cmp.setData({ drawsIndex: MOCK_DATA.draws.length - 1 });
    const next = cmp.find(".result-nav-next");
    expect(next.attributes("disabled")).toBe("disabled");
  });

  it("can navigate results - prev", () => {
    cmp.setData({ drawsIndex: 1 });
    const prev = cmp.find(".result-nav-prev");
    prev.trigger("click");
    expect(cmp.vm.drawsIndex).toBe(0);
  });

  it("disables prev button", () => {
    const prev = cmp.find(".result-nav-prev");
    expect(prev.attributes("disabled")).toBe("disabled");
  });
});
