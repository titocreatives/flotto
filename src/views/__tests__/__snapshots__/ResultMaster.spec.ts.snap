// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page ResultMaster matches snapshot 1`] = `
<div
  class="page result-eurojackpot"
>
  <section
    class="result-stage"
  >
    <div
      class="result-nav"
    >
      <button
        class="result-nav-prev"
        disabled="disabled"
      >
        
          ⟵
        
      </button>
       
      <div
        class="atom date f-date result-nav-date"
      >
        <span
          class="date-day"
        >
          13.
        </span>
         
        <span
          class="date-month"
        >
          
    Dezember
  
        </span>
         
        <span
          class="date-year"
        >
          2019
        </span>
      </div>
       
      <button
        class="result-nav-next"
      >
        
          ⟶
        
      </button>
    </div>
     
    <div
      class="o hero-numbers"
    >
      <div
        class="hero-numbers-grid"
      >
        <div
          class="hero-numbers-cell"
        >
          <figure
            class="ball as-default"
          >
            <span
              class="ball-number f-number"
            >
              25
            </span>
             
            <span
              class="ball-shadow"
            />
          </figure>
        </div>
        <div
          class="hero-numbers-cell"
        >
          <figure
            class="ball as-default"
          >
            <span
              class="ball-number f-number"
            >
              31
            </span>
             
            <span
              class="ball-shadow"
            />
          </figure>
        </div>
        <div
          class="hero-numbers-cell"
        >
          <figure
            class="ball as-default"
          >
            <span
              class="ball-number f-number"
            >
              38
            </span>
             
            <span
              class="ball-shadow"
            />
          </figure>
        </div>
        <div
          class="hero-numbers-cell"
        >
          <figure
            class="ball as-default"
          >
            <span
              class="ball-number f-number"
            >
              49
            </span>
             
            <span
              class="ball-shadow"
            />
          </figure>
        </div>
        <div
          class="hero-numbers-cell"
        >
          <figure
            class="ball as-default"
          >
            <span
              class="ball-number f-number"
            >
              50
            </span>
             
            <span
              class="ball-shadow"
            />
          </figure>
        </div>
        <div
          class="hero-numbers-cell"
        >
          <figure
            class="ball as-additional"
          >
            <span
              class="ball-number f-number"
            >
              5
            </span>
             
            <span
              class="ball-shadow"
            />
          </figure>
        </div>
        <div
          class="hero-numbers-cell"
        >
          <figure
            class="ball as-additional"
          >
            <span
              class="ball-number f-number"
            >
              10
            </span>
             
            <span
              class="ball-shadow"
            />
          </figure>
        </div>
      </div>
    </div>
  </section>
</div>
`;
